File: Java/src/main/java/parrot/NorwegianBlueParrot.java
package parrot;

public class NorwegianBlueParrot extends Parrot {

    protected double voltage;
    protected boolean isNailed;

    public NorwegianBlueParrot(double voltage, boolean isNailed) {
        super();
        this.voltage = voltage;
        this.isNailed = isNailed;
    }

    @Override
    public double getSpeed() {
        return (isNailed) ? 0 : getBaseSpeed(voltage);
    }

    @Override
    public String getCry() {
        return voltage > 0 ? "Bzzzzzz" : "...";
    }

    protected double getBaseSpeed(double voltage) {
        return Math.min(24.0, voltage * getBaseSpeed());
    }
}

File: Kotlin/src/main/kotlin/parrot/NorwegianBlueParrot.kt
package parrot

class NorwegianBlueParrot : Parrot {
    constructor(numberOfCoconuts: Int, voltage: Double, isNailed: Boolean) : super(
        type = ParrotTypeEnum.NORWEGIAN_BLUE,
        numberOfCoconuts,
        voltage,
        isNailed
    )

    override val speed: Double = if (isNailed) 0.0 else getBaseSpeed(voltage)
    override val cry: String =
        if (voltage > 0) "Bzzzzzz"
        else "..."
}